src/pyhf/infer/calculators.py

```diff
@@ -699,13 +699,25 @@ def distributions(self, poi_test, track_progress=None):
         tensorlib, _ = get_backend()
         sample_shape = (self.ntoys,)
 
-        signal_pars = self.pdf.config.suggested_init()
-        signal_pars[self.pdf.config.poi_index] = poi_test
+        signal_pars = fixed_poi_fit(
+            poi_test,
+            self.data,
+            self.pdf,
+            self.init_pars,
+            self.par_bounds,
+            self.fixed_params,
+        )
         signal_pdf = self.pdf.make_pdf(tensorlib.astensor(signal_pars))
```

Comment:
As `fixed_poi_fit` returns a Tensor then the `astensor` call here is redundant and can now be removed. :+1: